pkg/controllers/management/clustertemplate/clustertemplate_rbac.go

```diff
@@ -0,0 +1,115 @@
+package clustertemplate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/controllers/management/globalnamespacerbac"
+	"github.com/rancher/rancher/pkg/namespace"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	creatorIDAnn = "field.cattle.io/creatorId"
+	ctLabel      = "io.cattle.field/clusterTemplateId"
+)
+
+type clusterTemplateController struct {
+	clusterTemplates              v3.ClusterTemplateInterface
+	clusterTemplateLister         v3.ClusterTemplateLister
+	clusterTemplateRevisionLister v3.ClusterTemplateRevisionLister
+	managementContext             *config.ManagementContext
+}
+
+type clusterTemplateRevisionController struct {
+	clusterTemplateRevisions      v3.ClusterTemplateRevisionInterface
+	clusterTemplateRevisionLister v3.ClusterTemplateRevisionLister
+	clusterTemplateLister         v3.ClusterTemplateLister
+	managementContext             *config.ManagementContext
+}
+
+func registerRbacControllers(ctx context.Context, mgmt *config.ManagementContext) {
+	ct := clusterTemplateController{
+		managementContext:             mgmt,
+		clusterTemplates:              mgmt.Management.ClusterTemplates(""),
+		clusterTemplateLister:         mgmt.Management.ClusterTemplates("").Controller().Lister(),
+		clusterTemplateRevisionLister: mgmt.Management.ClusterTemplateRevisions("").Controller().Lister(),
+	}
+	ct.clusterTemplates.AddHandler(ctx, "cluster-template-rbac-controller", ct.sync)
+
+	ctr := clusterTemplateRevisionController{
+		clusterTemplateRevisions:      mgmt.Management.ClusterTemplateRevisions(""),
+		clusterTemplateRevisionLister: mgmt.Management.ClusterTemplateRevisions("").Controller().Lister(),
+		clusterTemplateLister:         mgmt.Management.ClusterTemplates("").Controller().Lister(),
+	}
+	ctr.clusterTemplateRevisions.AddHandler(ctx, "cluster-template-rev-rbac-controller", ctr.sync)
+}
+
+func (ct *clusterTemplateController) sync(key string, clusterTemplate *v3.ClusterTemplate) (runtime.Object, error) {
+	if clusterTemplate == nil || clusterTemplate.DeletionTimestamp != nil {
+		return nil, nil
+	}
+	metaAccessor, err := meta.Accessor(clusterTemplate)
+	if err != nil {
+		return clusterTemplate, err
+	}
+	creatorID, ok := metaAccessor.GetAnnotations()[globalnamespacerbac.CreatorIDAnn]
+	if !ok {
+		return clusterTemplate, fmt.Errorf("clusterTemplate %v has no creatorId annotation", metaAccessor.GetName())
+	}
+
+	if err := globalnamespacerbac.CreateRoleAndRoleBinding(globalnamespacerbac.ClusterTemplateResource, clusterTemplate.Name,
+		globalnamespacerbac.RancherManagementAPIVersion, creatorID, []string{globalnamespacerbac.RancherManagementAPIVersion},
+		clusterTemplate.UID,
+		clusterTemplate.Spec.Members, ct.managementContext); err != nil {
+		return nil, err
+	}
```

Comment:
Should we call globalnamespacerbac.CreateRoleAndRoleBinding only when obj.Spec.Members has changed? Looks like we call on every sync

Reply:
I guess no way to differentiate the updates?

Reply:
@prachidamle yeah I don't think so. The function `CreateRoleAndRoleBinding` does compare the subjects of rolebindings before actually update rolebindings, so we should be fine I guess?

Reply:
as discussed offline with @alena1108 @mrajashree - we cannot know if Members got updates, but have to do a full sync - we should use cache lookups in globalnamespacerbac.CreateRoleAndRoleBinding as much as possible